internal/tablespace_spec.go

```diff
@@ -0,0 +1,147 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/tinsane/tracelog"
+	"github.com/wal-g/wal-g/utility"
+	"path/filepath"
+)
+
+const (
+	BasePrefix  = "base_prefix"
+	Tablespaces = "tablespaces"
+)
+
+// The mandatory keys for this map are "base_prefix" and "tablespaces".
+// "base_prefix" contains Location of pg_data folder.
+// "tablespaces" contains array of keys, which also happen to be names of tablespace folders.
+// The rest keys should be these names of tablespace folders and values should be TablespaceLocation structs.
+type TablespaceSpec map[string]interface{}
+
+type TablespaceLocation struct {
+	Location string `json:"loc"`
+	Symlink  string `json:"link"`
+}
+
+func logInconsistentState(message string) {
+	tracelog.WarningLogger.Printf("TablespaceSpecification has inconsistent state:\n%s", message)
+}
+
+func (spec TablespaceSpec) requireBasePrefix() (string, bool) {
+	basePrefix, ok := spec.BasePrefix()
+	if !ok {
+		logInconsistentState("Base prefix not set while working with tablespaces.")
```

Comment:
I decided that i don't really need requireBasePrefix.
I don't really need HasBasePrefix also, because BasePrefix gives that information as a second parameter.